bigquery_etl/shredder/config.py

```diff
@@ -156,6 +156,7 @@ def fields(self) -> tuple[str, ...]:
 
 DELETE_TARGETS: DeleteIndex = {
     client_id_target(table="fenix_derived.new_profile_activation_v1"): FENIX_SRC,
+    client_id_target(table="fenix_derived.firefox_android_clients_v1"): FENIX_SRC,
```

Comment:
It's possible, but it's not currently supported. Tasks are ordered by `partition_sort_key DESC, sql_table_id ASC`:
https://github.com/mozilla/bigquery-etl/blob/61b9b2ffe3f22ec183f4b9edd1597203e1f5b8ee/bigquery_etl/shredder/delete.py#L525-L528
https://github.com/mozilla/bigquery-etl/blob/61b9b2ffe3f22ec183f4b9edd1597203e1f5b8ee/bigquery_etl/shredder/delete.py#L380-L390

for example the order if both tables are over 10TiB, and the only tables in `fenix_derived` being shredded:
```
... # other tables <= 10TiB
... # other tables > 10TiB by partition for 20230728
fenix_derived.firefox_android_clients_v1$20230728
fenix_derived.new_profile_activation_v1$20230728
... # other tables > 10TiB by partition for 20230728
... # other tables > 10TiB by partition for 20230727
fenix_derived.firefox_android_clients_v1$20230727
fenix_derived.new_profile_activation_v1$20230727
... # other tables > 10TiB by partition for 20230727
```
if both tables are under 10TiB, they will be shredded as whole tables instead of per partition:
```
... # other tables <= 10TiB
fenix_derived.firefox_android_clients_v1
fenix_derived.new_profile_activation_v1
... # other tables <= 10TiB
... # other tables > 10TiB by partition
```

Reply:
some things to note:
- shredder always starts at least one day further in the past than daily ETL. so if today is 2023-07-28, then daily ETL will be running for 2023-07-27, and if a shredder run started today it would only process dates up to 2023-07-26.
- shredder processes every deletion request twice, just in case it conflicts with a backfill operation or the like, because it's effectively impossible to guarantee a particular query will "win" when two queries write to the same partition at the same time, without requiring some sort of coordination outside of BQ.

Reply:
> @relud we would want `fenix_derived.new_profile_activation_v1` to always be shredded first or at the same time as `fenix_derived.firefox_android_clients_v1`. Is it possible?

> It's possible, but it's not currently supported.

if we have to accomplish this, it should be done by adding a "priority" field to delete targets, and assigning `firefox_android_clients_v1` a reduced priority compared to whatever default priority `new_profile_activation_v1` would get, and including a comment on why the priority is needed for these tables.

A decision would also need to be made about whether this priority is more or less important the the partition ordering. if more important, then `firefox_android_clients_v1` would run after *all other tables and partitions*. if less important and `firefox_android_clients_v1` were less than 10TiB¹ while `new_profile_activation_v1` were greater than 10TiB¹, then the priority would have no impact because partition ordering puts whole-table tasks before per-partition tasks.

Also potentially relevant is that in slack @lucia-vargas-a has requested shredder `processing per-partition the tables that are base for KPIs`, which would make impacted tables ignore the 10TiB limit to induce a more convenient "paper trail" for identifying shredder KPI impact. https://mozilla.slack.com/archives/C4D5ZA91B/p1690457722299319?thread_ts=1690221165.742359&cid=C4D5ZA91B

¹ The 10TiB limit is configured here: https://github.com/mozilla/bigquery-etl/blob/61b9b2ffe3f22ec183f4b9edd1597203e1f5b8ee/bigquery_etl/shredder/delete.py#L100-L109 and applied here https://github.com/mozilla/bigquery-etl/blob/61b9b2ffe3f22ec183f4b9edd1597203e1f5b8ee/bigquery_etl/shredder/delete.py#L360